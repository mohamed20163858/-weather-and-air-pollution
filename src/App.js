import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { fetchAllCaptialCities } from './redux/cities/cities';
import Navbar from './components/Navbar';
import Cities from './components/Cities';
import City from './components/city';

const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchAllCaptialCities());
  });

  return (
    <Router>
      <Navbar />
      <Routes>
        <Route path="/" element={<Cities />} />
        <Route path="/:id" element={<City />} />
      </Routes>
    </Router>
  );
};

export default App;
