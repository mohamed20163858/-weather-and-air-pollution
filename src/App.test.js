import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import store from './redux/configureStore';
import '@testing-library/jest-dom';
import App from './App';

function TestApp() {
  return (
    <Provider store={store}>
      <App />
    </Provider>
  );
}
it('renders correctly', () => {
  const tree = renderer
    .create(<TestApp />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('simulate navigating  through details link ', () => {
  // ARRANGE
  render(<TestApp />);

  // ACT
  userEvent.click(screen.getByTestId('nav-test-id'));
  // Arrange
  const result = screen.getByText('there is no country has this name please enter a valid country name');
  // ASSERT
  expect(result).toBeInTheDocument();
});
