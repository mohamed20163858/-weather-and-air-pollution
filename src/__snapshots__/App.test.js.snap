// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="navbar"
  >
    <a
      aria-current="page"
      className="logo active"
      data-testid="nav-test-id"
      href="/"
      onClick={[Function]}
    >
      <i
        className="fa-solid fa-house-user"
      />
    </a>
    <p>
      Weather and Air Pollution
    </p>
    <div
      className="left"
    >
      <i
        className="fa-solid fa-microphone"
      />
      <i
        className="fa-solid fa-gear"
      />
    </div>
  </div>,
  <div
    className="cards"
    data-testid="momo-id"
  >
    <div
      className="search"
    >
      <i
        className="fa-solid fa-magnifying-glass"
      />
      <input
        onKeyUp={[Function]}
        placeholder="search using country, city"
        type="text"
      />
    </div>
    <p>
      there is no country has this name please enter a valid country name
    </p>
  </div>,
]
`;
